### PR TITLE
add embedding model by default to distribution templates

### PR DESCRIPTION
**Pull request by dineshyv**
# What does this PR do?
Adds the sentence transformer provider and the `all-MiniLM-L6-v2` embedding model to the default models to register in the run.yaml for all providers.

## Test Plan
llama stack build --template together --image-type conda
 llama stack run ~/.llama/distributions/llamastack-together/together-run.yaml

### CODE REVIEW

File: docs/source/distributions/self_hosted_distro/meta-reference-quantized-gpu.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ The `llamastack/distribution-meta-reference-quantized-gpu` distribution consists
 | agents | `inline::meta-reference` |
 | datasetio | `remote::huggingface`, `inline::localfs` |
 | eval | `inline::meta-reference` |
-| inference | `inline::meta-reference-quantized` |
+| inference | `inline::meta-reference-quantized`, `remote::sentence-transformers` |
```
Comment (raghotham) [inline docs/source/distributions/self_hosted_distro/meta-reference-quantized-gpu.md:20]: why is it remote:: - isnt sentence-transformer inline?

File: llama_stack/templates/cerebras/cerebras.py [hunk 2]
```diff
@@ -8,10 +8,14 @@
 
 from llama_models.sku_list import all_registered_models
 
+from llama_stack.apis.models.models import ModelType
+
 from llama_stack.distribution.datatypes import ModelInput, Provider, ShieldInput
+from llama_stack.providers.inline.inference.sentence_transformers import (
+    SentenceTransformersInferenceConfig,
+)
```
Comment (raghotham) [inline llama_stack/templates/cerebras/cerebras.py:16]: is the idea to use sentence transformer initially and then work with providers to see which embedding models they have?

Comment (dineshyv) [reply llama_stack/templates/cerebras/cerebras.py:16]: This is mainly for back compat reasons since we will break user flows otherwise.

**Review by raghotham — APPROVED** [review]

File: llama_stack/providers/inline/inference/sentence_transformers/config.py [hunk 3]
```diff
@@ -7,4 +7,8 @@
 from pydantic import BaseModel
 
 
-class SentenceTransformersInferenceConfig(BaseModel): ...
+class SentenceTransformersInferenceConfig(BaseModel):
+
+    @classmethod
+    def sample_run_config(cls) -> "SentenceTransformersInferenceConfig":
```
Comment (ashwinb) [inline llama_stack/providers/inline/inference/sentence_transformers/config.py:13]: this signature is wrong. it should be `-> dict:`

**Review by ashwinb — APPROVED** [review]
This is great. I have one follow-up comment. You had indicated that this will still break with older run.yaml files naturally since the model is not registered. Can you see how exactly it breaks (what error shows up and what the backtrace is)? Then we can see how we can add a near-term hack potentially -- or at least a very clear error message.

File: llama_stack/providers/inline/inference/sentence_transformers/config.py (hunk 3 above)
Comment (dineshyv) [reply llama_stack/providers/inline/inference/sentence_transformers/config.py]: done.

**Comment by dineshyv** [comment]
Since existing run.yaml does not have an embedding model registered explicitly for `all-MiniLM-L6-v2`, the distribution does not know which provider to route to for this model. The stack trace is this:
```
Traceback (most recent call last):
  File "/Users/dineshyv/code/llama-stack/llama_stack/distribution/server/server.py", line 187, in endpoint
    return await maybe_await(value)
  File "/Users/dineshyv/code/llama-stack/llama_stack/distribution/server/server.py", line 151, in maybe_await
    return await value
  File "/Users/dineshyv/code/llama-stack/llama_stack/providers/utils/telemetry/trace_protocol.py", line 101, in async_wrapper
    result = await method(self, *args, **kwargs)
  File "/Users/dineshyv/code/llama-stack/llama_stack/distribution/routers/routing_tables.py", line 314, in register_memory_bank
    raise ValueError(f"Model {params.embedding_model} not found")
ValueError: Model all-MiniLM-L6-v2 not found
```

**Pull request merged**